dev_tools/Dockerfile

```diff
@@ -1,8 +1,18 @@
-FROM ubuntu
+FROM python:3.7-slim AS compile-image
 
 # Install dependencies.
+# rm -rf /var/lib/apt/lists/* cleans up apt cache. See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
 RUN DEBIAN_FRONTEND=noninteractive apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-     python3-pip python3-tk texlive-latex-base latexmk git emacs vim locales
+     python3-pip \
+     python3-tk \
+     texlive-latex-base \
+     latexmk \
+     git \
+     emacs \
```

Comment:
let's remove emacs and vim, it is not our responsibility to bundle these tools in my opinion.